blockchain/process.go

```diff
@@ -65,15 +65,21 @@ func (b *BlockChain) ProcessBlock(block *dcrutil.Block, flags BehaviorFlags) (in
 		return 0, ruleError(ErrDuplicateBlock, str)
 	}
 
+	blockHeader := &block.MsgBlock().Header
+	prevHash := &blockHeader.PrevBlock
+	isTreasuryEnabled, err := b.isTreasuryAgendaActiveByHash(prevHash)
+	if err != nil {
+		return 0, err
+	}
+
 	// Perform preliminary sanity checks on the block and its transactions.
-	err := checkBlockSanity(block, b.timeSource, flags, b.chainParams)
+	err = checkBlockSanity(block, b.timeSource, flags, b.chainParams,
+		isTreasuryEnabled)
```

Comment:
This is basically a no go, imo.  The entire point of `checkBlockSanity` is that is **context free**.  Adding the flag makes it rely on the state of the chain which will 100% break eventual multi-peer code and therefore this simply can't be here.

Reply:
Per our conversation I split the context-free functions into two different functions that do the context-free and contextual checks.

I kept the original name of the function with the `isTreasuryEnabled` flag before rigging up the code everywhere this is called.